### PR TITLE
Re-enable upgrade testing

### LINKED ISSUE #717: Re-enable upgrade testing
Upgrade testing is currently disabled.

We need to re-enable the upgrade test suite, hopefully re-using the Vagrant libvirt work that has been implemented for EFS and SSI testing.

### LINKED ISSUE #803: Upgrade EE 2.4.2.7 to IML 4.0.9.1 to IML 5 fails during migration
### Description 

`chroma-config setup` fails during migration phase.

### Repro

This system is an upgrade from EE 2.4.2.7 to IML 4.0.9.1 to IML 5

### Version

IML 5 (lustre-2-12 branch)
el7.6

### Diagnostics
```
# chroma-config setup -v
Starting setup...

DB already accessible
Creating database tables...
Syncing...
Creating tables ...
Installing custom SQL ...
Installing indexes ...
Installed 0 object(s) from 0 fixture(s)
Migrating...
Traceback (most recent call last):
  File "/usr/bin/chroma-config", line 10, in <module>
    load_entry_point('iml-manager==4.1.0.0', 'console_scripts', 'chroma-config')()
  File "/usr/share/chroma-manager/chroma_core/lib/service_config.py", line 941, in chroma_config
    errors = service_config.setup(username, password, ntpserver, check_db_space)
  File "/usr/share/chroma-manager/chroma_core/lib/service_config.py", line 660, in setup
    error = self._setup_database(check_db_space)
  File "/usr/share/chroma-manager/chroma_core/lib/service_config.py", line 542, in _setup_database
    self._syncdb()
  File "/usr/share/chroma-manager/chroma_core/lib/service_config.py", line 524, in _syncdb
    ManagementUtility(args).execute()
  File "/usr/lib/python2.7/site-packages/django/core/management/__init__.py", line 392, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/lib/python2.7/site-packages/django/core/management/base.py", line 242, in run_from_argv
    self.execute(*args, **options.__dict__)
  File "/usr/lib/python2.7/site-packages/django/core/management/base.py", line 285, in execute
    output = self.handle(*args, **options)
  File "/usr/lib/python2.7/site-packages/django/core/management/base.py", line 415, in handle
    return self.handle_noargs(**options)
  File "/usr/lib/python2.7/site-packages/south/management/commands/syncdb.py", line 103, in handle_noargs
    management.call_command('migrate', **options)
  File "/usr/lib/python2.7/site-packages/django/core/management/__init__.py", line 159, in call_command
    return klass.execute(*args, **defaults)
  File "/usr/lib/python2.7/site-packages/django/core/management/base.py", line 285, in execute
    output = self.handle(*args, **options)
  File "/usr/lib/python2.7/site-packages/south/management/commands/migrate.py", line 111, in handle
    ignore_ghosts = ignore_ghosts,
  File "/usr/lib/python2.7/site-packages/south/migration/__init__.py", line 200, in migrate_app
    applied_all = check_migration_histories(applied_all, delete_ghosts, ignore_ghosts)
  File "/usr/lib/python2.7/site-packages/south/migration/__init__.py", line 95, in check_migration_histories
    raise exceptions.GhostMigrations(ghosts)
south.exceptions.GhostMigrations: 

 ! These migrations are in the database but not on disk:
    <chroma_core: 0040_table_update_notify>
 ! I'm not trusting myself; either fix this yourself by fiddling
 ! with the south_migrationhistory table, or pass --delete-ghost-migrations
 ! to South to have it delete ALL of these records (this may not be good).
```
Re-running produces same error.

Possibly related to pull #798 ?

### LINKED ISSUE #882: 4.0 MountOrImportStep broken in 5.0
The `MountOrImportStep` has a case in 4.0 where the active_volume_node can be `None`

Screen Shot 2019-04-19 at 7 53 48 PM

The logic to describe this properly was eliminated in 5.0.

This means any lookup of a 4.x `MountOrImportStep` (used for unmount) will return a 500 error in 5.0

### LINKED ISSUE #883: Upgrade from IML 4.0.10.0 to 5.0.0 cannot upgrade lustre-all-dkms
Upgrading from 4.0.10 to 5.0.0 in managed mode does not work. 5.0 tries to install `lustre-zfs-dkms`, but this conflicts with `lustre-all-dkms`.

```
Loading mirror speeds from cached hostfile
 * base: mirror.keystealth.org
 * epel: mirrors.kernel.org
 * extras: mirrors.xtom.com
 * updates: mirrors.xtom.com
Package kernel-devel-3.10.0-957.1.3.el7_lustre.x86_64 already installed and latest version
Package pcs-0.9.165-6.el7.centos.1.x86_64 already installed and latest version
Package fence-agents-all-4.2.1-11.el7_6.7.x86_64 already installed and latest version
Package fence-agents-virsh-4.2.1-11.el7_6.7.x86_64 already installed and latest version
Package python2-iml-agent-4.1.5-0.1555767071.el7.noarch already installed and latest version
Resolving Dependencies
--> Running transaction check
---> Package chroma-agent-management.noarch 0:4.0.10.0-5057.el7 will be obsoleted
---> Package lustre-ldiskfs-zfs.x86_64 0:1-3.el7.centos will be updated
---> Package lustre-ldiskfs-zfs.x86_64 0:3-21.1.el7 will be an update
--> Processing Dependency: lustre-zfs-dkms for package: lustre-ldiskfs-zfs-3-21.1.el7.x86_64
---> Package lustre-resource-agents.x86_64 0:2.10.7-1.el7 will be installed
---> Package python2-iml-agent-management.noarch 0:4.1.5-0.1555767071.el7 will be obsoleting
--> Running transaction check
---> Package lustre-zfs-dkms.noarch 0:2.10.7-1.el7 will be installed
--> Processing Conflict: lustre-all-dkms-2.10.7-1.el7.noarch conflicts lustre-zfs-dkms
--> Processing Conflict: lustre-zfs-dkms-2.10.7-1.el7.noarch conflicts lustre-all-dkms
--> Finished Dependency Resolution
 You could try using --skip-broken to work around the problem
Error: lustre-all-dkms conflicts with lustre-zfs-dkms-2.10.7-1.el7.noarch
Error: lustre-zfs-dkms conflicts with lustre-all-dkms-2.10.7-1.el7.noarch
```

### PR DESCRIPTION
**Pull request by jgrund**
Fixes #717.
Fixes #803.
Fixes #882.
Fixes #883.

Add upgrade testing back to jenkins and ensure it works from 4.x -> 5.0

Signed-off-by: Joe Grund <jgrund@whamcloud.io>

### CODE REVIEW

File: tests/framework/integration/installation_and_upgrade/run_tests [hunk 1]
```diff
@@ -149,151 +120,72 @@ wait_for_nodes "$CLIENT_1" "rpm -q kmod-lustre-client || exit 0
...
 TESTS="$TESTS_DIR/../shared_storage_configuration/test_cluster_setup.py \
-       $TESTS_DIR/test_update_with_yum.py:TestYumUpdate.test_clean_linux_devices \
        $TESTS_DIR/test_create_filesystem.py:TestCreateFilesystem.test_create"
 
 # shellcheck disable=SC2086
 ssh root@$TEST_RUNNER "exec 2>&1; set -xe
 cd /usr/share/chroma-manager/
-unset http_proxy; unset https_proxy
-export UPGRADE_FROM_3=$UPGRADE_FROM_3
+export IML_4_INSTALLED=yup
```
Comment (johnsonw) [inline tests/framework/integration/installation_and_upgrade/run_tests]: Why not "yes"? Just curious.

File: tests/framework/integration/installation_and_upgrade/run_tests [hunk 2]
```diff
@@ -149,151 +120,72 @@ wait_for_nodes "$CLIENT_1" "rpm -q kmod-lustre-client || exit 0
...
-        for repo in $STORAGE_SERVER_REPOS; do
-            curl \"\$repo\" >> /etc/yum.repos.d/Intel-Lustre-Agent.repo
-        done
-    fi" | dshbak -c
-    if [ "${PIPESTATUS[0]}" != "0" ]; then
-        exit 1
-    fi
-fi
+# Install and setup new manager
+scp "$CHROMA_DIR"/chroma_support.repo "$CHROMA_DIR"/_topdir/RPMS/noarch/python2-iml-manager{,-cli,-libs}-5* "$CHROMA_DIR"/tests/utils/upgrade.exp root@"$CHROMA_MANAGER":/tmp
```
Comment (johnsonw) [inline tests/framework/integration/installation_and_upgrade/run_tests]: Should the `5` here be an environment variable?

File: tests/framework/utils/defaults.sh.in [hunk 3]
```diff
@@ -17,11 +17,7 @@ set_distro_vars() {
 
     if [ "$distro" = "el7" ]; then
         if ${upgrade_test}; then
-            if ${UPGRADE_FROM_3:-false}; then
-                export TEST_DISTRO_VERSION=${TEST_DISTRO_VERSION:-"7.3"}
-            else
-                export TEST_DISTRO_VERSION=${TEST_DISTRO_VERSION:-"7.4"}
-            fi
+            export TEST_DISTRO_VERSION=${TEST_DISTRO_VERSION:-"7.5"}
```
Comment (johnsonw) [inline tests/framework/utils/defaults.sh.in]: Why not 7.6?

File: tests/framework/integration/installation_and_upgrade/run_tests (hunk 1 above)
Comment (jgrund) [reply tests/framework/integration/installation_and_upgrade/run_tests]: It doesn't matter what's here, the test only checks for var existence.

Comment (johnsonw) [reply tests/framework/integration/installation_and_upgrade/run_tests]: ok sounds good

File: tests/framework/integration/installation_and_upgrade/run_tests (hunk 2 above)
Comment (jgrund) [reply tests/framework/integration/installation_and_upgrade/run_tests]: maybe at some point. It's not being used elsewhere from what I see.

File: tests/framework/utils/defaults.sh.in (hunk 3 above)
Comment (jgrund) [reply tests/framework/utils/defaults.sh.in]: This is the the *starting* distro. We go from IML 4 + EL 7.5, to IML 5 + EL 7.6

Comment (johnsonw) [reply tests/framework/utils/defaults.sh.in]: ok that makes sense.

**Pull request merged**